### PR TITLE
Issue #12113: Github action: GitHub Milestone and Upload Jar"

### LINKED ISSUE #12113: Github action "GitHub Milestone and Upload Jar" should do all github actionvities from release.sh
https://github.com/checkstyle/checkstyle/blob/master/.github/workflows/update_github.yml

should do all actions from in github:
https://github.com/checkstyle/checkstyle/blob/master/release.sh#L66
- `echo "Updating Github tag page"`
- `echo "Publishing 'all' jar to Github"`
- `echo "Close previous milestone at github"`
- `echo "Creation of new milestone ..."`
- `echo "Creation of issue in eclipse-cs repo ..."`
- `echo "Creation of issue in sonar-checkstyle repo ..."`

there should be no overlap with [publish_releasenotes_outside.yml](https://github.com/checkstyle/checkstyle/blob/master/.github/workflows/publish_releasenotes_outside.yml) as this action can do update for any release in past.

------

fix name of script - https://github.com/checkstyle/checkstyle/blob/02babf8a0ab715caebddf299b373bac31f2a9969/.ci/update-github.sh#L27 as it does not exists in repo.

-----

update-github.sh should build '-all' jar
https://github.com/checkstyle/checkstyle/blob/02babf8a0ab715caebddf299b373bac31f2a9969/release.sh#L50-L51
but for this action need to checkout to released tag (example `checkstyle-10.3.3`).

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Fixes #12113 

update-github-tag-page.sh

- `echo "Updating Github tag page"`

NEW: update-all-jar.sh

- `echo "Publishing 'all' jar to Github"`

NEW: creation-of-new-milestone.sh

- `echo "Close previous milestone at github"`
- `echo "Creation of new milestone ..."`

NEW: creation-of-issue-in-other-repos.sh

- `echo "Creation of issue in eclipse-cs repo ..."`
- `echo "Creation of issue in sonar-checkstyle repo ..."`

### CODE REVIEW

File: .ci/new-github-milestone-and-publish-jar.sh
Comment (romani) [inline .ci/new-github-milestone-and-publish-jar.sh:1]: is there any problem to put all in existing action and shell file ?

**Review by romani — CHANGES_REQUESTED** [review]
items

File: .ci/new-github-milestone-and-publish-jar.sh
Comment (Rahulkhinchi03) [reply .ci/new-github-milestone-and-publish-jar.sh:1]: No, but I thought to make release scripts more specialized rather than focusing on the generalization approach. But I can revert changes if this does not look good to you.

Comment (Rahulkhinchi03) [reply .ci/new-github-milestone-and-publish-jar.sh]: Instead of creating a github action, I have merged the scrips so we don't need a particular script now.

Comment (Rahulkhinchi03) [reply .ci/new-github-milestone-and-publish-jar.sh]: update PR description: https://github.com/checkstyle/checkstyle/pull/12119#issue-1353314309

File: .ci/update-github-milestone.sh
Comment (romani) [inline .ci/update-github-milestone.sh:1]: this shell is used in the middle of development process, not actually a release.

it will be required to run it to update version in pom and milestone to correct version.

Example:
1) now we have 10.3.4-SNAPSHOT in pom.
2) we merge some PR that is for issue that is marked as "breaking compatibility"
3) build in master fail (if we finish https://github.com/checkstyle/contribution/issues/648)
4) maintianer review failure in CI and confirm that we need to change version to 10.4-SNAPSHOT
5) maintainer run action to update version in pom and in github milestone
6) ready to merge new PR ..... release will be considered to be done in some time in future, may in few weeks.

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: .ci/update-github-milestone.sh
Comment (romani) [reply .ci/update-github-milestone.sh:1]: so means that we need to have new action for this activity.
and may be in future publish_releasenotes_outside.yml  be merged in this new action, or may be allways stay separate.
lets keep it separate for now.

I do not know why but some gut filling tell me to not mix most critical actions in github and release notes. I do not want to fail release if some problem appeared in release notes generation/publishing.

Comment (Rahulkhinchi03) [reply .ci/update-github-milestone.sh:1]: @romani The idea of making every action with specialised approach was my first thought but with review, I changed it to general.

But I am making a special action for all the above tasks or maybe we can create 2 shell scripts:

1. Closing and Creating new milestone.
    - Publishing Jar.

2. Creating issue in other Repo.

And running both of these scripts in one GitHub action?

Comment (Rahulkhinchi03) [reply .ci/update-github-milestone.sh:1]: PR description updated: https://github.com/checkstyle/checkstyle/pull/12119#issue-1353314309

Comment (romani) [reply .ci/update-github-milestone.sh:1]: I am ok with 2 scripts

File: .ci/creation-of-new-milestone.sh
Comment (romani) [inline .ci/creation-of-new-milestone.sh:1]: .ci/close-create-milestone.sh

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .ci/creation-of-new-milestone.sh
Comment (Rahulkhinchi03) [reply .ci/creation-of-new-milestone.sh]: Changes Done

File: .ci/update-github.sh
Comment (romani) [inline .ci/update-github.sh:28]: there is not such folder.
do checkout by tag `git checkout checkstyle-$TARGET_VERSION`

File: .github/workflows/new-milestone-and-issue-in-other-repo.yml
Comment (romani) [inline .github/workflows/new-milestone-and-issue-in-other-repo.yml:19]: fix name

File: .github/workflows/new-milestone-and-issue-in-other-repo.yml
Comment (romani) [inline .github/workflows/new-milestone-and-issue-in-other-repo.yml:5]: "R: Update Milestone, Upload '-all' jar, Create issues"

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: .ci/update-github.sh
Comment (Rahulkhinchi03) [reply .ci/update-github.sh:28]: I am doing the changes but I am not sure if we have a tag ( for eg: the current version, `10.3.4` ) but at this moment we have `10.3.3` which is latest tag

File: .github/workflows/new-milestone-and-issue-in-other-repo.yml
Comment (Rahulkhinchi03) [reply .github/workflows/new-milestone-and-issue-in-other-repo.yml]: Changes Done

File: .ci/update-github.sh
Comment (romani) [reply .ci/update-github.sh]: Release:prepare will create tag 10.3.4.
This action will execute a bit later , it should expect that tag exists.
This action should be tested on 10.3.3 (any previous tag). Imagine that release (tag creation and copy of jars) was done and now we catch-up on updates of items in GitHub to beautify GitHub content.

File: .github/workflows/new-milestone-and-issue-in-other-repo.yml
Comment (romani) [inline .github/workflows/new-milestone-and-issue-in-other-repo.yml:5]: Let go back to `R: Close/Create Milestone,..`

I do not see upload in action body.

File: .ci/update-github.sh
Comment (romani) [inline .ci/update-github.sh:33]: No need to update milestone.

It is script for action that happens before release process

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .ci/update-github.sh
Comment (Rahulkhinchi03) [reply .ci/update-github.sh]: Changes Done

File: .github/workflows/new-milestone-and-issue-in-other-repo.yml
Comment (Rahulkhinchi03) [reply .github/workflows/new-milestone-and-issue-in-other-repo.yml]: Changes Done

File: .github/workflows/new-milestone-and-issue-in-other-repo.yml
Comment (romani) [inline .github/workflows/new-milestone-and-issue-in-other-repo.yml:24]: Move it back to shell script and name script  as upload-all-jar.sh

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .github/workflows/new-milestone-and-issue-in-other-repo.yml
Comment (Rahulkhinchi03) [reply .github/workflows/new-milestone-and-issue-in-other-repo.yml]: Changes Done

File: .ci/update-github.sh
Comment (romani) [inline .ci/update-github.sh:30]: Move mvn build to .ci/upload-all-jar.sh

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .ci/update-github.sh
Comment (romani) [reply .ci/update-github.sh]: done

**Comment by romani** [comment]
I updated all scripts to be ok. I did not test them, I will test them after merge.

**Review by romani — APPROVED** [review]
this PR bring 2 actions:
one action all all steps
another action do only file upload (it is temporal action for testing, we will remove after testing)

**Pull request merged**
